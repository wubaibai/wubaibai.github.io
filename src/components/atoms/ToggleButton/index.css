.button {
	background: var(--standard);
	padding: 4px 10px;
	border-radius: 5px;
	border: none;
	color: var(--secondary);
}

.button-reverse {
	composes: button;
	background: var(--secondary);
	color: var(--standard);
}
