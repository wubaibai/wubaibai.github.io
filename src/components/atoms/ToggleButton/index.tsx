import React from 'react';

import { useBoolean } from 'util/hook';

import style from './index.css';

interface ToggleButtonProperty {
	onClose: () => void;
	onOpen: () => void;
	closeTitle?: string;
	openTitle?: string;
}

const ToggleButton: React.FC<ToggleButtonProperty> = ({
	onClose,
	onOpen,
	closeTitle = 'Close',
	openTitle = 'Open',
}) => {
	const [isOpen, { toggle }] = useBoolean({
		onTrue: onOpen,
		onFalse: onClose,
		defaultBoolean: true,
	});

	return (
		<button type="button" className={isOpen ? style.button : style.buttonReverse} onClick={toggle}>
			{isOpen ? closeTitle : openTitle}
		</button>
	);
};

export default ToggleButton;
