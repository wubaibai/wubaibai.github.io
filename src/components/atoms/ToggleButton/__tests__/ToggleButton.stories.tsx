import React from 'react';
import { text } from '@storybook/addon-knobs';
import { storiesOf } from '@storybook/react';
import { action } from '@storybook/addon-actions';

import ToggleButton from 'components/atoms/ToggleButton';

const stories = storiesOf('atoms/ToggleButton', module);

stories.add(
	'__interactive',
	() => (
		<ToggleButton
			onOpen={action('open')}
			onClose={action('close')}
			openTitle={text('Open Title', 'Open')}
			closeTitle={text('Close Title', 'Close')}
		/>
	),
	{ jest: 'ToggleButton' },
);
