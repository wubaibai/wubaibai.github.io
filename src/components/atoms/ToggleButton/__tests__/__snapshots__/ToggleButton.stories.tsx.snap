// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`__interactive 1`] = `
<DocumentFragment>
  <button
    class="button"
    type="button"
  >
    Close
  </button>
</DocumentFragment>
`;
